geode-core/src/main/java/org/apache/geode/internal/tcp/ClusterConnection.java

```diff
@@ -2909,13 +2950,13 @@ private boolean readMessageHeader(ByteBuffer peerDataBuffer) throws IOException
       readerShuttingDown = true;
       requestClose(String.format("Unknown P2P message type: %s",
           nioMessageTypeInteger));
-      return true;
+      return false;
```

Comment:
not obvious why this bool is flipped...

Reply:
The old meaning didn't make a lot of sense.  Returning _false_ when the header is successfully read?  No.  That's why I changed it.